### DISCUSSION TITLE
Reach an array in a composable through a middleware

### DISCUSSION THREAD

**Discussion by zupcode** [nuxt/nuxt#19515] [Questions] [answered]
Hello,
I'm stuck... I'm trying to reach through a global middleware an element of an array who's in a composable.

Some code to explain :

I have an object array in a composable.global.ts :
```
export const useMenuColorState = () => useState('menuColorState', () => ({
  bio: false,
  contact: false
}))
```
and when I try to reach for example `menuColorState.bio` in a middleware :

```
export default defineNuxtRouteMiddleware((to, from) => {
  const menuColorState = useState('menuColorState')
  console.log(menuColorState.bio.value);
}
```
it returns undefined

but if I do :
`console.log(menuColorState.value);`
I can see on the console :
```
Proxy { <target>: {…}, <handler>: {…} }
<target>: Object { bio: false, contact: false }
```
but I can't reach `bio `or `contact` like in a template `menuColorState.bio`

Is there a solution for that ?

**Comment by imlautaro** [comment, marked as answer]
Try with `menuColorState.value.bio` instead of `menuColorState.bio.value` 🤔

**Comment by zupcode** [comment]
I've found my mistake...
I was calling the useState in the composable on the wrong way...
`const menuColorState = useState('menuColorState')`
it seems to works only for one level states.

I just have to call it simply like in a composable or a page :
`const menuColorState = useMenuColorState()`
and now it works with
`menuColorState.value.bio`
as you said

Thanks for the answer !
